packages/pigeon/diff_tool.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+###############################################################################
+# A tool that helps you check the Pigeon output for a given file across
+# different versions of Pigeon.
+#
+# The comparison will be made between the currently checked out sha and the one
+# provided as an argument.
+#
+# usage: ./diff_tool.sh <sha for commit to test against> <path to pigeon file>
+###############################################################################
+
+xHash=$1
+pigeonPath=$2
+diffTool="diff -ru"
+gitTool="git -c advice.detachedHead=false"
+
+generate_everything() {
+  local inputPath=$1
+  local outputDir=$2
+  pub run pigeon \
+    --input $inputPath \
+    --dart_out $outputDir/dart.dart \
+    --java_out $outputDir/java.dart \
+    --objc_header_out $outputDir/objc.h \
+    --objc_source_out $outputDir/objc.m
+}
+
+yHash=$(git rev-parse HEAD)
+xDir=$(mktemp -d -t $xHash)
+yDir=$(mktemp -d -t $yHash)
+inputPath=$yDir/input.dart
```

Comment:
Does `mktemp` guarantee that it won't generate a path containing any spaces? I didn't see any specific guarantee in the docs, but maybe I missed it. If not, these all need to be quoted.

Reply:
Yes, I do believe that is a guarantee as long as you don't do something funky to change $TMPDIR.  Quoted anyways.